### ISSUE TITLE
beforeMessageBodyTransformed and afterMessageBodyTransformed are unusable

### ISSUE THREAD

**Issue by immae** [conversejs/converse.js#1809]
**Describe the bug**
In file `src/converse-message-view.js`, function `transformBodyText`, there are two triggers that are defined to help "transforming a chat message's body text" (quoting the doc).
However, it seems impossible to do such modification in the hooks, since the `text` variable is not passed as a reference but as a string, so any change happening there won’t have any consequences on the resulting text.

Did I miss something?

**Comment by jcbrand** [comment]
The _converse.MessageView object is also passed to event handlers.

So if you wanted to change the text, you could do something like:

`view.model.save({'text': 'new text'}, {'silent': true});`

**Comment by jcbrand** [comment]
https://github.com/conversejs/converse.js/blob/306748feb0a1d8222cb55db9e8ae00bdc4df7b3f/src/converse-message-view.js#L220

**Comment by immae** [comment]
Thanks, I wasn’t aware of the possibility for view.model.save. However, it looks like your `.model.save({'text': 'new text'}, {'silent': true});` has no effect, is it supposed to work as is?

**Comment by immae** [comment]
Actually, if I understand correctly, you meant `view.model.set('message', 'new text')`, but this will in result re-call the trigger, which ends up in an infinite call loop.

**Comment by jcbrand** [comment]
> but this will in result re-call the trigger, which ends up in an infinite call loop.

Which is why I'm passing in `{'silent': true}`.

**Comment by jcbrand** [comment]
> has no effect

Yeah... because the text isn't read from the model any more. I'll reopen the ticket so that this can be fixed.
